File: 0075-sort-colors/0075-sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let zeros = 0;
    let ones = 0;
    let twos = 0;

    for(let j=0; j<nums.length; j++){
            if(nums[j] === 0) zeros++;
            if(nums[j] === 1) ones++;
            if(nums[j] === 2) twos++;
        }
    
    let i=0;
    while(zeros--) nums[i++] = 0;
    while(ones--) nums[i++] = 1;
    while(twos--) nums[i++] = 2;

    return nums;

     
};

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
    // most optical solution with in-place
    int n = nums.size();
    int low=0, mid=0, high=n-1;

    while(mid <= high){ // 1
       if(nums[mid] == 0) {
             swap(nums[mid++], nums[low++]);
          }
       else if(nums[mid] == 2){  // 2
        swap(nums[mid], nums[high--]);
         }
       else { // 1
        mid++; 
        }
      }
    }
};